lib/openstack_query/query_executer.py

```diff
@@ -0,0 +1,113 @@
+import time
+import logging
+from typing import Callable, Optional, Dict, List, Any, Union, Type
+
+from openstack_query.runners.query_runner import QueryRunner
+from enums.cloud_domains import CloudDomains
+from enums.query.props.prop_enum import PropEnum
+
+from custom_types.openstack_query.aliases import (
+    OpenstackResourceObj,
+    ServerSideFilters,
+    ClientSideFilterFunc,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class QueryExecuter:
+    """
+    Helper class to handle executing the query - primarily performing 'run()' method
+    """
+
+    def __init__(self, prop_enum_cls: Type[PropEnum], runner: QueryRunner):
+        self._prop_enum_cls = prop_enum_cls
+        self.runner = runner
+        self._parser_func = None
```

Comment:
... I think we can simply do `prop_enum_cls.get_marker_prop_func()` here instead?